### PR TITLE
 Added hotkey configuration menu

### PR DESCRIPTION
**Pull request by yuupo123**
Not a c ++ programmer, but it might be useful

### CODE REVIEW

File: src/platform/windows/input/key.h [hunk 1]
```diff
@@ -37,6 +37,7 @@ struct Key {
     static Key mouseButton(SDL_MouseButtonEvent button);
     static Key controllerMove(SDL_ControllerAxisEvent axis);
     static Key controllerButton(SDL_ControllerButtonEvent button);
+    const char* Key::getName();
```
Comment (JaCzekanski) [inline src/platform/windows/input/key.h]: Remove `Key::`

**Review by JaCzekanski — CHANGES_REQUESTED** [review]

**Comment by JaCzekanski** [comment]
Looks great! Thanks for contributing.
Could you take a look at non-Windows builds? Looks like somethings wrong there:
```c++
../../../src/platform/windows/input/key.h:40:22: error: extra qualification on member 'getName'
    const char* Key::getName();
```

One thing I don't like is how Soft/Hard reset is handled, but to be honest, it's a limitation of the current key mapping system - `keyboard|F2` doesn't allow you to specify modifiers like Shift.

Also, it won't work if somebody was to map a hotkey to something else than a keyboard - for example, I could map `quick save/load` to a game controller, but that would fail due to hardcoded keyboard event check
```c++
if (event.key.keysym.sym == Key(config.hotkeys["quick_load"]).key)
```
There are some modifications needed to `SdlInputManager` to handle it, but for now, we can leave it as it.

File: src/platform/windows/gui/options/options.cpp [hunk 2]
```diff
@@ -192,18 +193,42 @@ void drawImage(const std::optional<Image> image, float w = 0.f, float h = 0.f) {
...
+    key = Key(key).getName();
+
+    if (hotkey) {
+        if (ImGui::Button(fmt::format("{}##{}", key, button).c_str(), ImVec2(150.f, 0.f))) {
+            currentButton = button;
+            inputManager->waitingForKeyPress = true;
+            ImGui::OpenPopup("Waiting for key...");
+        }
+        ImGui::SameLine();
+        ImGui::Text(tooltip);
```
Comment (JaCzekanski) [inline src/platform/windows/gui/options/options.cpp]:
```c++
/home/build/android/app/jni/../../../build/libs/avocado/../../../src/platform/windows/gui/options/options.cpp:228:21: error: format string is not a string literal (potentially insecure) [-Werror,-Wformat-security]
          ImGui::Text(tooltip);
```
Use 
```c++
ImGui::TextUnformatted
```

**Review by JaCzekanski — CHANGES_REQUESTED** [review]
Android build fails :(

File: src/platform/windows/gui/options/options.cpp (hunk 2 above)
Comment (yuupo123) [reply src/platform/windows/gui/options/options.cpp]: It looks like a small error (ImGui::Text (tooltip) -> ImGui::Text ("%s",tooltip) ?). But unfortunately now I can only build and test on windows.

Comment (JaCzekanski) [reply src/platform/windows/gui/options/options.cpp]: Both `ImGui::TextUnformatted(tooltip)` and `ImGui::Text ("%s",tooltip)` would work in this case.
Dunno why only Android NDK compiler has this check enabled.

**Comment by JaCzekanski** [comment]
Tested - works great :)

Ignore failing Android builds - I forgot to merge another branch where that was fixed.

**Pull request merged**